aten/src/ATen/native/LegacyDefinitions.cpp

```diff
@@ -51,15 +51,27 @@ Tensor& zero_(Tensor& self) {
 }
 
 Tensor & masked_fill_(Tensor& self, const Tensor & mask, Scalar value) {
-  return at::legacy::th::_th_masked_fill_(self, mask, value);
+  if (mask.dtype() == at::ScalarType::Byte) {
```

Comment:
nit: a comment about why you need to do this would be useful (just say something like we dispatch on `self` and TH is type-chcked, so we need different definitions.  This can be fixed by moving to ATen`.